scheds/rust/scx_p2dq/src/bpf/main.bpf.c

```diff
@@ -398,19 +409,37 @@ static s32 pick_idle_cpu(struct task_struct *p, struct task_ctx *taskc,
 	struct node_ctx *nodec;
 	struct llc_ctx *llcx;
 	bool interactive = is_interactive(taskc);
+	int llc_idle = 0;
 	s32 cpu = prev_cpu;
 
 	idle_cpumask = scx_bpf_get_idle_cpumask();
 	idle_smtmask = scx_bpf_get_idle_smtmask();
 
 	if (!idle_cpumask || !idle_smtmask || bpf_cpumask_empty(idle_cpumask))
-		goto out_put_cpumask;
+		goto found_cpu;
 
 	if (!(prev_cpuc = lookup_cpu_ctx(prev_cpu)) ||
 	    !(llcx = lookup_llc_ctx(prev_cpuc->llc_id)) ||
 	    !(nodec = lookup_node_ctx(prev_cpuc->node_id)) ||
-	    !llcx->cpumask)
-		goto out_put_cpumask;
+	    !llcx->cpumask || !llcx->tmp_cpumask)
+		goto found_cpu;
+
+	bpf_cpumask_and(llcx->tmp_cpumask, idle_cpumask, cast_mask(llcx->cpumask));
+	int nr_idle = bpf_cpumask_weight(idle_cpumask);
+	if (llcx->tmp_cpumask)
+		llc_idle = bpf_cpumask_weight(cast_mask(llcx->tmp_cpumask));
+
+
+	// Special handling of tasks with custom affinities
```

Comment:
Going to punt on the affinitized tasks for now as when trying to add a test around all the different CPU selection spots it got clunky.